build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.code-quality.gradle.kts

```diff
@@ -61,7 +61,7 @@ dependencies {
     codenarc("gradlebuild:code-quality-rules") {
         because("Provides the IntegrationTestFixturesRule implementation")
     }
-    codenarc("org.codenarc:CodeNarc:2.0.0")
+    codenarc("org.codenarc:CodeNarc:3.0.1")
```

Comment:
Do we really need this, given that `code-qualtiy-rules` already defined a dependency on `CodeNarc`?

Reply:
The dependency in `code-quality-rules` is `compileOnly`, which I think makes sense. So we need to set the right codenarc version here, I think.

Reply:
I don't have enough context/insights here. To me it seems like the codenarc rules need a codenarc runtime to execute. Though they shouldn't depend on the runtime, since you'd want to choose an arbitrary runtime. That is why my impression was that `compileOnly` is good.